pkg/odo/cli/list/component/list.go

```diff
@@ -190,7 +217,11 @@ func HumanReadableOutput(list api.ResourcesList) {
 			managedBy = text.Colors{text.FgBlue}.Sprintf(managedBy)
 		}
 
-		t.AppendRow(table.Row{name, componentType, mode, managedBy})
+		row := table.Row{name, componentType, mode, managedBy}
+		if feature.IsEnabled(ctx, feature.GenericRunOnFlag) {
+			row = append(row, comp.RunningOn)
```

Comment:
To be consistent in the human-readable form, can we display something like `None` when `comp.RunningOn` is empty?

```
$ ODO_EXPERIMENTAL_MODE=true odo list component
Experimental mode enabled for flag: --run-on. Use at your own risk. More details on https://odo.dev/docs/user-guides/advanced/experimental-mode
 ✓  Listing components from namespace 'default' [210ms]
 NAME                                       PROJECT TYPE  RUNNING IN   MANAGED       RUNNING ON 
 my-nodesjs-with-k8s-uri-with-var           nodejs        Dev, Deploy  odo (v3.3.0)  cluster    
 my-nodejs-app-with-composite-run-commands  nodejs        Dev          odo (v3.3.0)  cluster    
 * a-test-nodejs                            Node.js       None         Unknown                  

```